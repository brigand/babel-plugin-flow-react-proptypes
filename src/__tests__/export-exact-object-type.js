const babel = require('babel-core');
const content = `
// @flow
import React from 'react'
export type VendorProps = {|
  test: string,
|};
`;

it('export-exact-object-type', () => {
  const res = babel.transform(content, {
    babelrc: false,
    presets: ['env', 'stage-1', 'react'],
    plugins: ['syntax-flow', require('../')],
  }).code;
  expect(res).toMatchSnapshot();
});
