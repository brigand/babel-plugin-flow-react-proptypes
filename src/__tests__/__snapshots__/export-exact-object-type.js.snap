// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`export-exact-object-type 1`] = `
"'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var babelPluginFlowReactPropTypes_proptype_VendorProps = {
  test: require('prop-types').string.isRequired
};
if (typeof exports !== 'undefined') Object.defineProperty(exports, 'babelPluginFlowReactPropTypes_proptype_VendorProps', {
  value: babelPluginFlowReactPropTypes_proptype_VendorProps,
  configurable: true
});"
`;
